src/main/java/org/fogbowcloud/manager/core/threads/FulfilledMonitor.java

```diff
@@ -0,0 +1,175 @@
+                    } catch (OrderStateTransitionException e) {
+                        LOGGER.error("Error while trying to process the order " + order, e);
+                    }
+                } else {
+                    this.fulfilledOrdersList.resetPointer();
+                    LOGGER.info("There is no fulfilled order to be processed, sleeping for "
+                            + this.sleepTime + " milliseconds");
+                    Thread.sleep(this.sleepTime);
+                }
+            } catch (InterruptedException e) {
+                isActive = false;
+                LOGGER.warn("Thread interrupted", e);
+            } catch (Throwable e) {
+                LOGGER.error("Not expected error", e);
+            }
+        }
+    }
+
+    /**
+     * Get an instance for a fulfilled order. If that instance is not reachable
+     * then order state is set to failed.
+     *
+     * @param order {@link Order}
+     * @throws OrderStateTransitionException Could not remove order from list of fulfilled orders.
+     */
+    protected void processFulfilledOrder(Order order) throws OrderStateTransitionException {
+        synchronized (order) {
+            OrderState orderState = order.getOrderState();
+
+            if (orderState.equals(OrderState.FULFILLED)) {
+                LOGGER.info("Trying to get an instance for order [" + order.getId() + "]");
+
+                try {
+                    this.processInstance(order);
+                } catch (OrderStateTransitionException e) {
+                    LOGGER.error("Error while trying to get an instance for order: " + order, e);
+
+                    LOGGER.info("Transition [" + order.getId() + "] order state from fulfilled to failed");
+                    OrderStateTransitioner.transition(order, OrderState.FAILED);
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if instance state is failed and changes the order state to failed.
+     * Checks SSH connectivity if instance state is active and the order type is compute.
+     *
+     * @param order {@link Order}
+     */
+    protected void processInstance(Order order) throws OrderStateTransitionException {
+        InstanceProvider instanceProvider = this.getInstanceProviderForOrder(order);
+
+        OrderInstance orderInstance = instanceProvider.getInstance(order);
+        OrderType orderType = order.getType();
+
+        InstanceState instanceState = orderInstance.getState();
+
+        if (instanceState.equals(InstanceState.FAILED)) {
+            LOGGER.info("Instance state is failed for order [" + order.getId() + "]");
+            OrderStateTransitioner.transition(order, OrderState.FAILED);
+        } else if (instanceState.equals(InstanceState.ACTIVE) && orderType.equals(OrderType.COMPUTE)) {
+            LOGGER.info("Processing active compute instance for order [" + order.getId() + "]");
+
+            ComputeOrderInstance computeOrderInstance = (ComputeOrderInstance) orderInstance;
```

Comment:
The compute order instance has not the tunneling service informations, who has this information is the compute order instance from the Order.